fido/fido.py

```diff
@@ -215,15 +215,14 @@ def handle_timeout_errors(error):
         """
 
         if error.check(_twisted_web_client().ResponseNeverReceived):
-            if error.value.reasons[0].check(CancelledError):
-                raise crochet.TimeoutError(
-                    "Connection was closed by fido because the server took "
-                    "more than timeout={timeout} seconds to "
-                    "send the response".format(timeout=timeout)
-                )
+            raise HTTPTimeoutError(
+                "Connection was closed by fido because the server took "
+                "more than timeout={timeout} seconds to "
+                "send the response".format(timeout=timeout)
+            )
 
-        elif error.check(TwistedTimeoutError):
-            raise crochet.TimeoutError(
+        elif error.check(TwistedConnectError):
+            raise TCPConnectError(
                 "Connection was closed by Twisted Agent because the HTTP "
                 "connection took more than connect_timeout={connect_timeout} "
                 "seconds to establish.".format(connect_timeout=connect_timeout)
```

Comment:
Can you update the wording of this error since it covers more than just connection timeouts?